app/src/main/java/rec/games/pokemon/teambuilder/PokemonListAdapter.java

```diff
@@ -58,49 +82,66 @@ public int getItemCount()
 		return mPokemon.size();
 	}
 
+	public int getPokemonClickedId(int position)
+	{
+		Log.d(TAG, "position: " + position);
+		if(position > 0 && mPokemon != null)
+		{
+			Log.d(TAG, Integer.toString(mPokemon.getValue(position - 1).getId()));
+			return mPokemon.getValue(position - 1).getId(); //mPokemon ids start at 1
```

Comment:
With how I initially setup the ViewModel and LiveDataList, the list will never have nulls inside. But that doesn't stop one from using a bad index to get a null value or an exception.

At this point in time in the code, we can guarantee that the Pokemon is a instance of DeferredPokemonResource, but it may not be a PokemonResource. We would need to check for that if you want to use anything special from non-deferred class

Reply:
But you bring up a pretty good point of whether or not we should be assuming the LiveData objects contain null values

Reply:
Okay, that makes sense.